proxy/src/outbound.rs

```diff
@@ -133,7 +144,8 @@ where
 
         Buffer::new(balance, self.bind.executor())
             .map(|buffer| {
-                InFlightLimit::new(buffer, MAX_IN_FLIGHT)
+                let inflight = InFlightLimit::new(buffer, MAX_IN_FLIGHT);
+                Timeout::new(inflight, self.bind_timeout, &self.handle)
```

Comment:
I wonder if maybe `InFlightLimit` should instead wrap `Timeout`, so that if the request were to hit the limit, we didn't create a timer just to throw away immediately? 🤔

Reply:
Yeah, that seems reasonable. I wasn't sure where the right place in the stack for the timeout was...